core/src/main/java/com/linecorp/armeria/server/VirtualHost.java

```diff
@@ -395,6 +396,10 @@ public Routed<ServiceConfig> findServiceConfig(RoutingContext routingCtx, boolea
                 maybeSetRoutingResult(routingCtx, routed);
                 return routed;
             case NOT_MATCHED:
+                if (routingCtx.method() == HttpMethod.HEAD) {
+                    return findServiceConfig(routingCtx.withMethod(HttpMethod.GET));
```

Comment:
`useFallbackService` should be propagated. 
```suggestion
                    return findServiceConfig(routingCtx.withMethod(HttpMethod.GET), useFallbackService);
```

Reply:
Oh.. thank you for finding. 😱